src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -1879,8 +1879,8 @@ void updateForm(Function<LambdaForm, LambdaForm> updater) {
                 if (oldForm != newForm) {
                     assert (newForm.customized == null || newForm.customized == this);
                     newForm.prepare(); // as in MethodHandle.<init>
+                    UNSAFE.storeStoreFence(); // properly publish newForm
```

Comment:
I agree `fullFence` was misplaced. Not sure if `storeStore` is sufficient here, if we exposed the old instance somehow. It is safer to use release. And I think an even cleaner way would be to call `Unsafe.putReferenceRelease` below, which solves this as well.

Reply:
Thanks Aleksey, makes sense. Let's use `putReferenceRelease` then. There are a few other, existing uses of `UNSAFE.storeStoreFence` in the libraries code, especially in similar places in `java/lang/invoke/`. Maybe these should be double checked.